CONTRIBUTING.md

```diff
@@ -44,170 +64,249 @@ not to be able to mint negative values
--  A pull request must pass all automated checks before being merged. At a minimum, the code must be formatted, passing all tests, as well as having no outstanding `clippy` lints.
--  If you feel like your code isn't ready to merge, but you want the maintainers to look through it, e.g. to help, create a draft pull request.
--  A pull request cannot be merged without two approving reviews from the active maintainers.
--  Each pull request will automatically notify the code owners. An up to date list of current maintainers can be found in [MAINTAINERS.md](MAINTAINERS.md).
--  If during the preparation of your pull request your branch got out of date **DO NOT USE THE GitHub merge branch button**. You should instead rebase your commits on top of the recently merged changes. In the interest of making this process easier for everyone, try not to have more than a handful of commits, and avoid re-using feature branches.
--  Follow the [Rust Style Guide](#rust-style-guide)
--  Follow the [Git Style Guide](#git-style-guide)
--  Follow the [Documentation Style Guide](#documentation-style-guide)
--  When working with **PRs from forks** check [this manual](https://help.github.com/articles/checking-out-pull-requests-locally)
+## Pull Request Etiquette
 
-### Pull request titles
-Internally, to generate changelogs we parse the titles of all the merged pull requests. Thus in order to pass the *`check-PR-title`* check, you should do the following:
+Please [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo) the [repository](https://github.com/hyperledger/iroha/tree/iroha2-dev) and [create a feature branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository) for your contributions. When working with **PRs from forks**, check [this manual](https://help.github.com/articles/checking-out-pull-requests-locally).
 
-Put the type of pull request into square brackets as the first part of the title. This can be `feature`, `fix`, `ci`, `documentation`, and `refactor`. Example:
-```
-[feature] #1623: implement a `RawGenesisBlockBuilder`
-```
+Working on code contribution:
+- Follow the [Rust Style Guide](#rust-style-guide) and the [Documentation Style Guide](#documentation-style-guide).
+- Ensure that the code you've written is covered by tests. If you fixed a bug, please turn the minimum working example that reproduces the bug into a test.
+
+Committing your work:
+- Follow the [Git Style Guide](#git-style-guide).
+- Squash your commits [either before](https://www.git-tower.com/learn/git/faq/git-squash/) or [during the merge](https://rietta.com/blog/github-merge-types/).
+- If during the preparation of your pull request your branch got out of date, **DO NOT USE THE GitHub merge branch button**.
```

Comment:
Could mention that while using the default merge branch button is advised against, the same button has a drop down option for rebasing and that one does its job perfectly fine.

Reply:
At the time of writing the `update with Rebase` wasn't an option. Now it is. So the information needs to be updated. 

There's also hope that we can convince Hyperledger to allow some branches to be `merged` rather than `rebased` particularly for large works.

Reply:
updated